lib/src/storage_file_api.dart

```diff
@@ -4,302 +4,275 @@ import 'package:storage_client/src/fetch.dart';
-  /// [expiresIn] The number of seconds until the signed URL expires. For example, `60` for a URL which is valid for one minute.
-  Future<StorageResponse<String>> createSignedUrl(
+  /// [toPath] is the new file path, including the new file name. For example
+  /// `folder/image-copy.png`.
+  Future<String> copy(String fromPath, String toPath) async {
+    final options = FetchOptions(headers: headers);
+    final response = await storageFetch.post(
+      '$url/object/copy',
+      {
+        'bucketId': bucketId,
+        'sourceKey': fromPath,
+        'destinationKey': toPath,
+      },
+      options: options,
+    );
+    return (response as Map<String, dynamic>)['message'] as String;
+  }
+
+  /// Create signed URL to download file without requiring permissions. This URL
+  /// can be valid for a set number of seconds.
+  ///
+  /// [path] is the file path to be downloaded, including the current file
+  /// names. For example: `createdSignedUrl('folder/image.png')`.
+  ///
+  /// [expiresIn] is the number of seconds until the signed URL expire. For
+  /// example, `60` for a URL which are valid for one minute.
+  Future<String> createSignedUrl(
     String path,
     int expiresIn,
   ) async {
-    try {
-      final _path = _getFinalPath(path);
-      final options = FetchOptions(headers: headers);
-      final response = await fetch.post(
-        '$url/object/sign/$_path',
-        {'expiresIn': expiresIn},
-        options: options,
-      );
-      if (response.hasError) {
-        return StorageResponse(error: response.error);
-      } else {
-        final signedUrl =
-            '$url${(response.data as Map<String, dynamic>)['signedURL']}';
-        return StorageResponse<String>(data: signedUrl);
-      }
-    } catch (e) {
-      return StorageResponse(error: StorageError(e.toString()));
-    }
+    final _path = _getFinalPath(path);
+    final options = FetchOptions(headers: headers);
+    final response = await storageFetch.post(
+      '$url/object/sign/$_path',
+      {'expiresIn': expiresIn},
+      options: options,
+    );
+    final signedUrlPath = (response as Map<String, dynamic>)['signedURL'];
+    final signedUrl = '$url$signedUrlPath';
+    return signedUrl;
+  }
+
+  /// Create signed URLs to download files without requiring permissions. These
+  /// URLs can be valid for a set number of seconds.
+  ///
+  /// [path] is the file paths to be downloaded, including the current file
+  /// names. For example: `createdSignedUrl(['folder/image.png', 'folder2/image2.png'])`.
+  ///
+  /// [expiresIn] is the number of seconds until the signed URLs expire. For
+  /// example, `60` for URLs which are valid for one minute.
+  Future<List<String>> createSignedUrls(
+    List<String> paths,
+    int expiresIn,
+  ) async {
+    final options = FetchOptions(headers: headers);
+    final response = await storageFetch.post(
+      '$url/object/sign/$bucketId',
+      {'expiresIn': expiresIn},
```

Comment:
Do we need to pass the `paths` as the body as well here?
```suggestion
      {
        'expiresIn': expiresIn,
        'paths': paths,
      },
```

https://github.com/supabase/storage-js/blob/main/src/lib/StorageFileApi.ts#L256